### PR TITLE
fix: wrapped native asset resolve to their appropriate native asset o…

### PR DESCRIPTION
**Pull request by JackShort**
* wrapped assets now resolve to the appropriate native asset. for instance when going to wETH details it will show as ETH

### CODE REVIEW

File: src/pages/TokenDetails/index.tsx [hunk 1]
```diff
@@ -135,7 +135,7 @@ export default function TokenDetails() {
     <TokenDetailsLayout>
       {tokenAddress && (
         <>
-          <TokenDetail address={tokenAddress} />
+          <TokenDetail address={tokenAddress} connectedChainId={connectedChainId} />
```
Comment (vm) [inline src/pages/TokenDetails/index.tsx:138]: you can do `useWeb3React` in TokenDetail rather than needing to pass in connectedChainId here

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 2]
```diff
@@ -199,12 +206,22 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
...
 
-  if (!token || !token.name || !token.symbol) {
+  if (!token || !token.name || !token.symbol || !connectedChainId) {
     return <LoadingTokenDetail />
   }
 
-  const tokenName = tokenDetailData.name
-  const tokenSymbol = tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
+  const wrappedNativeCurrency = WRAPPED_NATIVE_CURRENCY[connectedChainId]
+  const isWrappedNativeToken = wrappedNativeCurrency && wrappedNativeCurrency.address === token.address
```
Comment (vm) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:214]: nit: `wrappedNativeCurrency?.address === token.address`

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 3]
```diff
@@ -174,9 +175,15 @@ export function AboutSection({ address, tokenDetailData }: { address: string; to
...
+export default function LoadedTokenDetail({
+  address,
+  connectedChainId,
+}: {
+  address: string
+  connectedChainId: number | undefined
+}) {
   const token = useToken(address)
-  const currency = useCurrency(address)
+  let currency = useCurrency(address)
```
Comment (vm) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:186]: do you know if we intend to keep useCurrency? seems hard to follow

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 4]
```diff
@@ -199,12 +206,22 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
...
-  const tokenSymbol = tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
+  const wrappedNativeCurrency = WRAPPED_NATIVE_CURRENCY[connectedChainId]
+  const isWrappedNativeToken = wrappedNativeCurrency && wrappedNativeCurrency.address === token.address
+
+  if (isWrappedNativeToken) {
+    currency = nativeOnChain(connectedChainId)
+  }
+
+  const tokenName = isWrappedNativeToken && currency ? currency.name : tokenDetailData.name
+  const tokenSymbol =
```
Comment (vm) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:221]: this is getting really hard to follow... why can't you use the tokenDetailData stuff anymore?

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 5]
```diff
@@ -199,12 +206,22 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
...
+    isWrappedNativeToken && currency
+      ? currency.symbol
+      : tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
```
Comment (vm) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:224]: can we skip the uppercase thing here and move it to css? that will help simplify

**Review by vm — CHANGES_REQUESTED** [review]

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 4 above)
Comment (JackShort) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:221]: I think they still want the wETH tokenDetail stuff but the ETH symbols, name, and logo

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 3 above)
Comment (JackShort) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:186]: I think the point is to get the appropriate logo

File: src/pages/TokenDetails/index.tsx (hunk 1 above)
Comment (JackShort) [reply src/pages/TokenDetails/index.tsx:138]: yeah but considering we are already getting it from the parent - 1 child down isn't that bad

Comment (vm) [reply src/pages/TokenDetails/index.tsx:138]: The page should not be concerned with the implementation details of TokenDetail component though.

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 3 above)
Comment (vm) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:186]: Oh wild. Sounds good

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 4 above)
Comment (vm) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:221]: oh boy sounds good

File: src/components/Tokens/TokenDetails/TokenDetail.tsx (hunk 5 above)
Comment (vm) [reply src/components/Tokens/TokenDetails/TokenDetail.tsx:224]: can you name a variable for `tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol` then?

**Review by cmcewen — APPROVED** [review]
looks like a merge conflict to resolve
